lib/string.js

```diff
@@ -91,19 +91,30 @@ internals.String = class extends Any {
         });
     }
 
-    regex(pattern, name) {
+    regex(pattern, config) {
 
         Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
 
         pattern = new RegExp(pattern.source, pattern.ignoreCase ? 'i' : undefined);         // Future version should break this and forbid unsupported regex flags
 
-        return this._test('regex', pattern, function (value, state, options) {
+        const patternIsInvalidated = (typeof config === 'object' && Hoek.reach(config, 'invalidate'));
```

Comment:
Why use hoek to access it since it's a direct property and you just checked it's an object ?